extensions/amp-subscriptions/0.1/amp-subscriptions.js

```diff
@@ -480,29 +480,27 @@ export class SubscriptionService {
   /**
    * Re authorizes a platform
    * @param {!SubscriptionPlatform} subscriptionPlatform
-   * @return {!Promise}
    */
   reAuthorizePlatform(subscriptionPlatform) {
     this.platformStore_.getAvailablePlatforms()
         .forEach(platform => {
           platform.reset();
           this.platformStore_.resetEntitlementFor(
               platform.getServiceId());
+          this.fetchEntitlements_(platform);
         });
     this.renderer_.toggleLoading(true);
-    return this.fetchEntitlements_(subscriptionPlatform).then(() => {
-      this.subscriptionAnalytics_.serviceEvent(
-          SubscriptionAnalyticsEvents.PLATFORM_REAUTHORIZED,
-          subscriptionPlatform.getServiceId()
-      );
-      // deprecated event fired for backward compatibility
-      this.subscriptionAnalytics_.serviceEvent(
-          SubscriptionAnalyticsEvents.PLATFORM_REAUTHORIZED_DEPRECATED,
-          subscriptionPlatform.getServiceId()
-      );
-      this.platformStore_.reset();
-      this.startAuthorizationFlow_();
-    });
+    this.subscriptionAnalytics_.serviceEvent(
+        SubscriptionAnalyticsEvents.PLATFORM_REAUTHORIZED,
+        subscriptionPlatform.getServiceId()
+    );
+    // deprecated event fired for backward compatibility
+    this.subscriptionAnalytics_.serviceEvent(
+        SubscriptionAnalyticsEvents.PLATFORM_REAUTHORIZED_DEPRECATED,
+        subscriptionPlatform.getServiceId()
+    );
+    this.platformStore_.reset();
```

Comment:
I think this should go as a first line. Probably not essential for the current exact code, but would be more correct as a flow.

Reply:
Agree, done